src/librustc_typeck/check/mod.rs

```diff
@@ -5255,12 +5209,8 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
 
         let tcx = self.tcx;
 
-        let res = match self.rewrite_self_ctor(res, span) {
-            Ok(res) => res,
-            Err(ErrorReported) => return (tcx.types.err, res),
-        };
         let path_segs = match res {
-            Res::Local(_) | Res::Upvar(..) => Vec::new(),
+            Res::Local(_) | Res::Upvar(..) | Res::SelfCtor(_) => vec![],
```

Comment:
This is good, FWIW, `Local | Upvar` became just `Local` in between the `beta` and `master` branches.